app/controllers/general_shopping_lists_controller.rb

```diff
@@ -0,0 +1,16 @@
+class GeneralShoppingListsController < ApplicationController
+  before_action :find_user
+
+  def index
+    @user = current_user
+    @food = @user.foods.left_outer_joins(:recipe_foods).where(recipe_foods: { id: nil }).includes(:recipe_foods)
+    @food_counts = @food.count
+    @food_prices = @food.sum(:price)
+  end
```

Comment:
- [ ] @Munyabelden and @Fahim-Khan-P great work on your shopping list,however, am unable to generate a shopping list whenever I click on the generate button I end up with the result below, kindly ensure your controller is generating the shopping list and it can seen on the shopping list page
![image](https://github.com/Munyabelden/Rails_Capstone/assets/61408860/3c86f1eb-62d5-4bd4-802c-b7fd92ba3803)

Reply:
Hi dear @danmainah,
Thank you for your great advice😉😉
But I think that the Shopping list will be empty because of your food list, If you have food on the food list it will be not empty as I tested it in the below video for a better understanding😚😚

https://github.com/Munyabelden/Rails_Capstone/assets/74261465/451b19bd-b5f4-478c-aee2-a2f985edd7d3

### **But, I am not sure that it will be complete, So I am waiting for your advice if it's not complete🤨**